tests/src/test/fdc3.broadcast.ts

```diff
@@ -1,126 +1,189 @@
 
-    let channelsAppContext = {
-      type: "channelsAppContext",
-      reverseMethodCallOrder: false,
-      contextBroadcasts: {
-        instrument: true,
-        contact: false,
-      },
-      useAppChannel: false,
-      broadcastMultipleItems: false,
-      broadcastExecutionComplete: false,
-    };
-
     it("Broadcast method is callable", async () => {
       await window.fdc3.broadcast({
         type: "fdc3.instrument",
         id: { ticker: "AAPL" },
       });
     });
 
-    describe("User channels", () => {
+    describe("System channels", () => {
       beforeEach(async () => {
         await unsubscribeListeners();
         await window.fdc3.leaveCurrentChannel();
-        resetChannelsAppContext();
       });
 
       afterEach(async () => {
-        await broadcastSystemChannelCloseWindow();
+        await closeChannelsAppWindow();
       });
 
       it("Should receive context when adding a listener then joining a user channel before app B broadcasts context to the same channel", async () => {
         const errorMessage = `\r\nSteps to reproduce:\r\n- Add fdc3.instrument context listener to app A\r\n- App A joins channel 1\r\n- App B joins channel 1\r\n- App B broadcasts fdc3.instrument context${documentation}`;
 
         return new Promise(async (resolve, reject) => {
-          //Add context listener to app A
-          listener = await window.fdc3.addContextListener(null, (context) => {
-            expect(context.type).to.be.equal("fdc3.instrument", errorMessage);
-            resolve();
-          });
+          //Listen for when ChannelsApp execution is complete
+          const resolveExecutionCompleteListener = waitForContext(
+            "executionComplete",
+            await window.fdc3.getOrCreateChannel("app-control")
+          );
+
+          //Add context listener
+          listener = await window.fdc3.addContextListener(
+            null,
+            async (context) => {
+              expect(context.type).to.be.equals(
+                "fdc3.instrument",
+                errorMessage
+              );
+              resolve();
+              return;
+            }
+          );
 
           validateListenerObject(listener);
 
-          //App A joins channel 1
+          //Join system channel 1
           await joinChannel(1);
 
-          //Open ChannelsApp app. ChannelsApp joins channel 1, then broadcasts context
-          window.fdc3.open("ChannelsApp", channelsAppContext);
+          const channelsAppCommands = [
+            commands.joinSystemChannelOne,
+            commands.broadcastInstrumentContext,
+          ];
 
-          //if no context received throw error
-          await wait();
+          //Open ChannelsApp then execute commands in order
+          await window.fdc3.open(
+            "ChannelsApp",
+            buildChannelsAppContext(channelsAppCommands, true)
+          );
+
+          //wait for ChannelsApp to execute
+          await resolveExecutionCompleteListener;
```

Comment:
We've got a race condition here. On some runs the message its listening for has already been sent when the listener is added.  This doesn't matter as much on a user channel as you receive the last context on it when you join (although that could be from the last test...), on an app channel you do not (but could call `channel.getCurrentContext`) to retrieve it.

To resolve... add the getCurrentContext call, but also add a data field to the context with a test id of some sort so that we know its completed that particular test?

I_may_ have a go at doing this before you get back @Joe-Dunleavy. If so I'll need to move your PR onto the main repo where we can get at it...